R/dataframe-frame.R

```diff
@@ -2065,3 +2065,107 @@ dataframe__sample <- function(
     self$`_df`$sample_n(n$`_s`, with_replacement, shuffle, seed)
   })
 }
+
+#' Unstack a long table to a wide form without doing an aggregation
+#'
+#' This can be much faster than a pivot, because it can skip the grouping phase.
+#'
+#' @param ... <[`dynamic-dots`][rlang::dyn-dots]> Column name(s), selector(s) or
+#' expressions to include in the operation. If `NULL` (default), use all
+#' columns.
+#' @param step Number of rows in the unstacked frame.
+#' @param how Direction of the unstack. Must be one of `"vertical"` or
+#' `"horizontal"`.
+#' @param fill_values Fill values that don’t fit the new size with this value.
+#'
+#' @inherit as_polars_df return
+#' @examples
+#' df <- pl$DataFrame(x = LETTERS[1:8], y = 1:8)$with_columns(
+#'   z = pl$int_ranges(pl$col("y"), pl$col("y") + 2, dtype = pl$UInt8)
+#' )
+#' df
+#'
+#' df$unstack(step = 4, how = "vertical")
+#' df$unstack(step = 2, how = "horizontal")
+#' df$unstack(cs$numeric(), step = 5, fill_values = 0)
+dataframe__unstack <- function(
+  ...,
+  step,
+  how = c("vertical", "horizontal"),
+  fill_values = NULL
+) {
+  wrap({
+    check_dots_unnamed()
+    if (!(is_scalar_integerish(step) && isTRUE(step > 0L))) {
+      abort("`step` must be a single positive integer-ish value")
+    }
+    how <- arg_match0(how, values = c("vertical", "horizontal"))
+
+    dots <- list2(...)
+    if (length(dots) == 0) {
+      df <- self
+    } else {
+      for (i in seq_along(dots)) {
+        if (!is_string(dots[[i]]) && !is_polars_selector(dots[[i]]) && !is_polars_expr(dots[[i]])) {
+          abort("`...` only accepts column names or column selectors.")
+        }
+      }
+      df <- self$select(!!!dots)
```

Comment:
`parse_into_list_of_expressions()` works on selectors, but then `!!!dots` would error. I think using `list2()` and checking in the loop that we have column names, selectors, or exprs is fine.